book/src/rfcs/benchmarking.md

```diff
@@ -0,0 +1,109 @@
+
+## Parameters
+
+We may be interested in parameterizing some benchmark, for example benchmarking verification time but doing it with different circuits sizes, that would show how performance scales with the size of the circuit.
+The introduction of parameters doesn't cause any fundamental change to the benchmarks, but it can considerably increase the cost of running them, making them more the kind of benchmarks that we may not want to run every single time in CI.
+
+## Benchmarks
+
+Next the classes of benchmarks to consider, they are different combinations of metrics, layers and consumers.
+  
+| benchmarks               | repository    | metric   | for external use | for internal use | regression test | noise sensitive  |
+| ------------------------ | ------------- | -------- | ---------------- | ---------------- | --------------- | ---------------- |
+| kimchi time              | proof-systems | time     | yes              | yes              | yes             | yes              |
+| kimchi time flamegraph   | proof-systems | time     | no               | yes              | no              | no               |
+| kimchi instructions      | proof-systems | time     | no               | yes              | yes             | no               |
+| kimchi memory            | proof-systems | memory   | yes              | yes              | yes             | no               |
+| kimchi memory flamegraph | proof-systems | memory   | no               | yes              | no              | no               |
+| snarky time              | mina/snarky*  | time     | yes              | yes              | yes             | yes              |
+| snarky memory            | mina/snarky*  | memory   | yes              | yes              | yes             | no               |
+| pickles time             | mina/pickles* | time     | yes              | yes              | yes             | yes              |
+| pickles memory           | mina/pickles* | memory   | yes              | yes              | yes             | no               |
+
+\* these have to be in Mina now because pickles is in Mina, and while snarky has it's repo, to use it requires a few things still in Mina, if it were possible change that it would make these benchmarks easier and probably cheaper.
+
+### Kimchi time
+
+One of the simplest measuring running time, can be an specific case or allow more flexibility by introducing parameters like for example different circuit sizes. We already have most of the infrastructure for this with criterion. Would ideally have a dedicated computer, considering that there aren't that many PRs in proof-systems the requirements shouldn't be that high.
+
+### Kimchi time flamegraph
+
+This can be considered a variant of the previous, some more research is required but ideally can be generated from the already existing benchmarks with little to no changes.
```

Comment:
```suggestion
This can be considered a variant of the previous. Some more research is required but this information ideally can be generated from the already existing benchmarks with little to no changes.
```

Reply:
Link to previous section